### PR TITLE
fix: stop iOS focus-zoom on mobile text fields

### PR DESCRIPTION
**Pull request by zeval**
iOS Safari zoomed the page in on every text-field focus, making the app feel like a webpage rather than an app. Fields now stay put on focus.

## Important Changes

- `globals.css` adds an `@media (any-pointer: coarse)` rule forcing form controls to 16px — iOS only auto-zooms focused fields under 16px (our `text-sm`/`text-xs` controls were 12–14px). `any-pointer` so it also covers hybrid touch devices; the decorative code-block language `<select>` is included so its picker can't trigger zoom either. `!important` to beat more-specific editor selectors.

## Validation

- `pnpm --filter @kandev/web typecheck` / `lint` — pass
- `e2e/tests/mobile-zoom.spec.ts` (Pixel 5 project): asserts the context is coarse-pointer and focused fields compute ≥16px — passing in CI across all E2E shards
- Manually verified on a real iPhone: tapping the New Task title no longer zooms

## Possible Improvements

Low risk. 16px fields are slightly denser inside the fixed `h-7` inputs on touch; intended tradeoff. Casual pinch-zoom is intentionally left enabled — blocking it on iOS would require a JS gesture handler that also disables legitimate accessibility zoom (WCAG 1.4.4).

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.

### CODE REVIEW

File: apps/web/app/globals.css [hunk 1]
```diff
@@ -202,6 +202,20 @@
...
+   < 16px. Force 16px on coarse-pointer (touch) devices to stop focus-zoom.
+   Kept unlayered on purpose so it overrides Tailwind's text-size utilities
+   (e.g. text-sm/text-xs) — moving it into @layer base would let those win
+   and silently break the fix. */
+@media (pointer: coarse) {
+  input,
+  textarea,
+  select,
+  [contenteditable] {
+    font-size: 16px;
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css]: Fixed in dc8ff3e0 — narrowed the selector to `[contenteditable="true"], [contenteditable="plaintext-only"]` so `contentEditable={false}` decorations (mention chips, code-block views) are no longer forced to 16px.

[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css]: Fixed in dc8ff3e0 — added `!important` to the coarse-pointer rule so more-specific selectors like `.tiptap pre .code-block-language` (0.625rem) can no longer keep a focusable control under 16px on touch. iOS focus-zoom is purely a function of computed font-size, so forcing 16px there is correct.

File: apps/web/e2e/tests/mobile-zoom.spec.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const mobile = new MobileKanbanPage(testPage);
+    await mobile.goto();
+
+    const content = await testPage.locator('meta[name="viewport"]').getAttribute("content");
+    expect(content).toContain("maximum-scale=1");
+  });
+
+  test("form fields render at >= 16px to prevent iOS focus-zoom", async ({ testPage }) => {
+    const mobile = new MobileKanbanPage(testPage);
+    await mobile.goto();
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/e2e/tests/mobile-zoom.spec.ts:11]: Verified this isn't an issue: in Playwright Test the mobile-chrome project's `...devices["Pixel 5"]` IS applied to `browser.newContext()`, so the testPage context resolves to a coarse pointer. Confirmed empirically — in that context `matchMedia("(pointer: coarse)").matches === true` (and `ontouchstart in window`). Added an explicit guard assertion in dc8ff3e0 so the font-size check fails loudly if a future fixture change ever drops touch emulation.

File: apps/web/index.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="color-scheme" content="light dark" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1, viewport-fit=cover" />
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/index.html]: Intentional, informed product decision. The user explicitly asked to block casual pinch-zoom (the free pinch made the UI feel like a webpage, not an app). `maximum-scale=1` does **not** disable OS-level accessibility zoom — iOS triple-tap / AssistiveTouch magnification and Android system magnification operate independently of the page viewport, so low-vision users retain full zoom. The iOS focus-zoom bug is fixed separately by the 16px coarse-pointer rule, not this attribute. Documented the tradeoff in a comment next to the meta tag (dc8ff3e0). Resolving as by-design.

[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/index.html]: Intentional, informed product decision. The user explicitly asked to block casual pinch-zoom (the free pinch made the UI feel like a webpage, not an app). `maximum-scale=1` does **not** disable OS-level accessibility zoom — iOS triple-tap / AssistiveTouch magnification and Android system magnification operate independently of the page viewport, so low-vision users retain full zoom. The iOS focus-zoom bug is fixed separately by the 16px coarse-pointer rule, not this attribute. Documented the tradeoff in a comment next to the meta tag (dc8ff3e0). Resolving as by-design.

File: apps/web/app/globals.css [hunk 4]
```diff
@@ -202,6 +202,23 @@
...
+   win on specificity and keep the field under 16px. Only true editing hosts
+   are targeted — `[contenteditable="false"]` decorations (TipTap mention
+   chips, code-block views) are deliberately excluded. */
+@media (pointer: coarse) {
+  input,
+  textarea,
+  select,
+  [contenteditable="true"],
+  [contenteditable="plaintext-only"] {
+    font-size: 16px !important;
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css:223]: Good catch — fixed in 4d77f533. Scoped the rule to `select:not(.code-block-language)` so the decorative language picker keeps its 0.625rem size on touch. Native `<select>`s open a picker rather than zooming the viewport, so they don't need the 16px floor anyway; the comment now reflects this.

File: apps/web/e2e/tests/mobile-zoom.spec.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    // Guard: the 16px rule is gated on `@media (pointer: coarse)`. The
+    // mobile-chrome project (Pixel 5) emulates touch, so the testPage context
+    // resolves to a coarse pointer. Assert it here so a future fixture change
+    // that drops touch emulation fails loudly instead of silently skipping the
+    // rule under test.
+    const coarse = await testPage.evaluate(() => matchMedia("(pointer: coarse)").matches);
+    expect(coarse).toBe(true);
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/e2e/tests/mobile-zoom.spec.ts]: Verified empirically that this is not the case: the `mobile-chrome` project's Pixel 5 device IS applied to `browser.newContext()` in @playwright/test. Ran a throwaway spec that creates the context exactly like `testPage` (`browser.newContext({ baseURL })`) under `--project=mobile-chrome`: `matchMedia('(pointer: coarse)').matches === true`, `'ontouchstart' in window === true`. The guard assertion added in dc8ff3e0 (`expect(coarse).toBe(true)`) now enforces this in CI, and the existing mobile-* specs already rely on the same inheritance. So the coarse-pointer path is exercised and the font-size assertion is meaningful.

File: apps/web/index.html [hunk 6]
```diff
@@ -3,7 +3,12 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="color-scheme" content="light dark" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
+    <!-- maximum-scale=1 intentionally blocks casual page pinch-zoom for an app-like
+         feel. This does NOT disable OS-level accessibility zoom (iOS triple-tap /
+         AssistiveTouch, Android magnification), which is independent of the page
+         viewport. Focus-zoom is handled separately by the 16px coarse-pointer rule
+         in globals.css, not by this attribute. -->
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1, viewport-fit=cover" />
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/index.html]: Intentional product decision (documented at the meta tag). Blocking casual pinch-zoom was an explicit requirement — the free pinch made the UI feel like a webpage rather than an app. OS-level magnification (iOS triple-tap / AssistiveTouch, Android system magnifier) is independent of the page viewport and remains fully available to low-vision users, and the focus-zoom bug is handled by the 16px rule, not this attribute. Keeping as-is by design.

File: apps/web/e2e/tests/mobile-zoom.spec.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+// fixes: the 16px coarse-pointer form-control rule (globals.css) that stops
+// focus-zoom, and the maximum-scale=1 viewport meta (index.html) that blocks
+// casual page pinch-zoom.
+test.describe("Mobile zoom hardening", () => {
+  test("viewport meta pins maximum-scale=1", async ({ testPage }) => {
+    const mobile = new MobileKanbanPage(testPage);
+    await mobile.goto();
+
+    const content = await testPage.locator('meta[name="viewport"]').getAttribute("content");
+    expect(content).toContain("maximum-scale=1");
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/e2e/tests/mobile-zoom.spec.ts]: Fixed in 16192f05 — the assertion now splits the viewport content on `,` and checks for the exact `maximum-scale=1` token, so a typo like `maximum-scale=10` would fail.

File: apps/web/app/globals.css [hunk 8]
```diff
@@ -202,6 +202,24 @@
...
+   !important because some controls are styled by more specific selectors that
+   would otherwise win on specificity. Only true editing hosts are targeted:
+   `[contenteditable="false"]` decorations (TipTap mention chips, code-block
+   views) are excluded, and the decorative `.code-block-language` <select> is
+   left at its compact size — native selects open a picker rather than zooming
+   the viewport, so they don't need the 16px floor. */
+@media (pointer: coarse) {
+  input,
+  textarea,
+  select:not(.code-block-language),
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css]: Agreed — reverted the exemption in 10b158c7; the `.code-block-language` <select> is now under the 16px floor on coarse-pointer devices, so tapping the language picker no longer hits the focus-zoom path. (OpenCode had earlier flagged the 16px size as a regression, but that only holds on desktop — the rule is coarse-pointer-only, so desktop keeps the compact 0.625rem, and on touch a larger tap target is an improvement, not a regression.)

File: apps/web/app/globals.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css:223]: Follow-up: reverted this exemption in 10b158c7. Codex correctly pointed out the native <select> can still trigger iOS focus-zoom under 16px. The 16px size only applies on coarse-pointer (touch) — desktop keeps the compact 0.625rem — and a 16px tap target on touch is an improvement over the 10px one, so there's no real mobile regression.

File: apps/web/app/globals.css [hunk 9]
```diff
@@ -202,6 +202,25 @@
...
+/* iOS Safari auto-zooms a focused form control whose computed font-size is
+   < 16px (native <select> included). Force 16px on coarse-pointer (touch)
+   devices to stop focus-zoom. !important because some controls are styled by
+   more specific selectors that would otherwise win on specificity — e.g. the
+   compact `.tiptap pre .code-block-language` picker (0.625rem). Bumping that
+   picker to 16px only happens on touch, where a larger tap target is an
+   improvement; desktop (fine pointer) keeps the compact size. Only true
+   editing hosts get the contenteditable rule — `[contenteditable="false"]`
+   decorations (mention chips, code-block views) are excluded. */
+@media (pointer: coarse) {
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css]: Fixed in 7bdde52f — switched the media query to `@media (any-pointer: coarse)` so the 16px floor also covers hybrid devices whose primary pointer is fine (touchscreen laptop + mouse). Updated the e2e guard assertion to match.

File: apps/web/index.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/index.html]: Agreed and addressed in 42a8b62d — removed `maximum-scale=1` entirely. You're right that iOS 10+ ignores it for pinch-zoom, so it was a no-op on the primary target while failing WCAG 1.4.4 audits. The actual user-facing complaint (text fields zooming on focus) is fully handled by the 16px `any-pointer: coarse` rule, which stands on its own. Reliable iOS pinch suppression would require a JS gesture path with real accessibility cost; we deliberately chose not to disable user zoom.

[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/index.html]: Intentional — we're deliberately not adding a JS pinch-zoom guard. Blocking pinch on iOS requires intercepting `gesturestart`/`touchmove`, which also disables legitimate accessibility zoom (WCAG 1.4.4) — a tradeoff we chose not to make. Casual pinch-zoom stays enabled; the scope here is the focus-zoom fix (the 16px any-pointer rule), which is the actual user-facing bug. The PR title/description have been updated to reflect this narrowed scope.

File: apps/web/app/globals.css [hunk 10]
```diff
@@ -202,6 +202,28 @@
...
+   more specific selectors that would otherwise win on specificity — e.g. the
+   compact `.tiptap pre .code-block-language` picker (0.625rem). Bumping that
+   picker to 16px only happens on touch, where a larger tap target is an
+   improvement; desktop (fine pointer) keeps the compact size. Only true
+   editing hosts get the contenteditable rule — `[contenteditable="false"]`
+   decorations (mention chips, code-block views) are excluded.
+   `any-pointer` (not `pointer`) so the floor also applies on hybrid devices
+   whose primary pointer is fine (e.g. a touchscreen laptop with a mouse),
+   where a touch tap can still focus a field. */
+@media (any-pointer: coarse) {
```
[the remark this answers is not in the document]
Comment (zeval) [reply apps/web/app/globals.css:217]: Already done — the PR title is now "fix: stop iOS focus-zoom on mobile text fields" and the description's Important Changes section no longer references maximum-scale (updated right after this note landed). The squash-merge changelog entry will accurately reflect that only the focus-zoom fix shipped; casual pinch-zoom is intentionally left enabled.
